File: 育成中コンテンツ/広告/アドフリくん/adfurikunSDK_Android_moviereward_3.20.0/SampleJava/app/src/main/java/sample/adfurikunsdk/jp/samplejava/SimpleAppOpenAd.java
package sample.adfurikunsdk.jp.samplejava;

import android.app.Activity;
import android.os.Bundle;
import android.view.View;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import jp.tjkapp.adfurikunsdk.moviereward.AdfurikunAppOpenAd;
import jp.tjkapp.adfurikunsdk.moviereward.AdfurikunAppOpenAdListener;
import jp.tjkapp.adfurikunsdk.moviereward.AdfurikunMovieError;
import jp.tjkapp.adfurikunsdk.moviereward.AppOpenAdData;

public class SimpleAppOpenAd extends Activity {

    /* アプリ起動時広告インスタンス */
    private AdfurikunAppOpenAd mAppOpenAd;

    /* 広告リスナー */
    private AdfurikunAppOpenAdListener mListener = new AdfurikunAppOpenAdListener() {
        @Override
        public void onPrepareSuccess(@Nullable String appId) {
            Toast.makeText(getApplicationContext(),
                    "広告の準備が完了しました。", Toast.LENGTH_LONG).show();
        }

        @Override
        public void onPrepareFailure(@Nullable String appId, @Nullable AdfurikunMovieError error) {
            int errorCode = error != null ? error.getErrorCode() : 0;
            Toast.makeText(getApplicationContext(),
                    "広告の準備が失敗しました。(" + errorCode + ")", Toast.LENGTH_LONG).show();
        }

        @Override
        public void onStartPlaying(@NonNull AppOpenAdData data) {
            Toast.makeText(getApplicationContext(),
                    "広告の再生を開始しました。(" + data.getAdnetworkKey() + ":" + data.getAdnetworkName() + ")", Toast.LENGTH_LONG).show();
        }

        @Override
        public void onFailedPlaying(@NonNull AppOpenAdData data, @Nullable AdfurikunMovieError error) {
            Toast.makeText(getApplicationContext(),
                    "広告の再生が中断しました。(" + data.getAdnetworkKey() + ":" + data.getAdnetworkName() + ")", Toast.LENGTH_LONG).show();
        }

        @Override
        public void onAdClose(@NonNull AppOpenAdData data) {
            Toast.makeText(getApplicationContext(),
                    "広告を閉じました。(" + data.getAdnetworkKey() + ":" + data.getAdnetworkName() + ")", Toast.LENGTH_LONG).show();
        }
    };

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_app_open_ad);
        setTitle(getIntent().getStringExtra("title"));

        // ボタンを押すと広告を読み込む
        findViewById(R.id.load_ad_btn).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                if (mAppOpenAd != null) {
                    mAppOpenAd.load(3000);
                }
            }
        });

        // ボタンを押すと広告を表示する
        findViewById(R.id.show_ad_btn).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                if (mAppOpenAd != null) {
                    if (mAppOpenAd.isPrepared()) {
                        mAppOpenAd.play(SimpleAppOpenAd.this);
                    } else {
                        Toast.makeText(getApplicationContext(), "広告の準備中です。", Toast.LENGTH_LONG).show();
                    }
                }
            }
        });

        mAppOpenAd = new AdfurikunAppOpenAd(AdfurikunConst.ADFURIKUN_APP_OPEN_AD_APPID, this);
        mAppOpenAd.setAdfurikunAppOpenAdListener(mListener);
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        if (mAppOpenAd != null) {
            mAppOpenAd.destroy();
        }
    }
}

File: 広告/アドフリくん/adfurikunSDK_Android_moviereward_3.20.0/SampleKotlin/app/src/main/java/jp/glossom/sample/samplekotlin/SimpleAppOpenAd.kt
package jp.glossom.sample.samplekotlin

import android.app.Activity
import android.os.Bundle
import android.widget.Button
import android.widget.Toast
import jp.tjkapp.adfurikunsdk.moviereward.*

/**
 * アプリ起動時広告用サンプル
 */
class SimpleAppOpenAd : Activity() {

    /** アプリ起動時広告インスタンス */
    private var mAppOpenAd: AdfurikunAppOpenAd? = null

    /** 広告リスナー */
    private val mListener = object : AdfurikunAppOpenAdListener {
        override fun onPrepareSuccess(appId: String?) {
            Toast.makeText(applicationContext, "広告の準備が完了しました。", Toast.LENGTH_LONG).show()
        }

        override fun onPrepareFailure(appId: String?, error: AdfurikunMovieError?) {
            Toast.makeText(applicationContext, "広告の準備が失敗しました。(${error?.errorCode})", Toast.LENGTH_LONG).show()
        }

        override fun onStartPlaying(data: AppOpenAdData) {
            Toast.makeText(applicationContext, "広告の再生を開始しました。(${data.adnetworkKey}:${data.adnetworkName})", Toast.LENGTH_LONG).show()
        }

        override fun onFailedPlaying(data: AppOpenAdData, error: AdfurikunMovieError?) {
            Toast.makeText(applicationContext, "広告の再生が中断しました。(${data.adnetworkKey}:${data.adnetworkName})", Toast.LENGTH_LONG).show()
        }

        override fun onAdClose(data: AppOpenAdData) {
            Toast.makeText(applicationContext, "広告を閉じました。(${data.adnetworkKey}:${data.adnetworkName})", Toast.LENGTH_LONG).show()
        }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_app_open_ad)
        title = intent.getStringExtra("title")

        // ボタンを押すと広告を読み込む
        findViewById<Button>(R.id.load_ad_btn).setOnClickListener {
            mAppOpenAd?.load(3000)
        }

        // ボタンを押すと広告を表示する
        findViewById<Button>(R.id.show_ad_btn).setOnClickListener {
            mAppOpenAd?.let {
                if (it.isPrepared) {
                    it.play(this)
                } else {
                    Toast.makeText(applicationContext, "広告の準備中です。", Toast.LENGTH_LONG).show()
                }
            }
        }

        // アプリ起動時広告のインスタンスを生成する
        mAppOpenAd = AdfurikunAppOpenAd(AdfurikunConst.ADFURIKUN_APP_OPEN_AD_APPID, this)
        mAppOpenAd?.setAdfurikunAppOpenAdListener(mListener)
    }

    override fun onDestroy() {
        mAppOpenAd?.destroy()
        super.onDestroy()
    }
}
